pages/solana.tsx

```diff
@@ -0,0 +1,347 @@
+import styles from "../styles/solana.module.css";
+import Image from "next/image";
+import AffiliateImage from "../public/visuals/affiliate.webp";
+import Button from "../components/UI/button";
+import StarknetIcon from "../components/UI/iconsComponents/icons/starknetIcon";
+import Wallets from "../components/UI/wallets";
+import { useAccount, useContractWrite } from "@starknet-react/core";
+import ProgressBar from "../components/UI/progressBar";
+import { NextPage } from "next";
+import { WalletMultiButton } from "@solana/wallet-adapter-react-ui";
+import { useWallet } from "@solana/wallet-adapter-react";
+import { Call } from "starknet";
+import TxConfirmationModal from "../components/UI/txConfirmationModal";
+import { NotificationType, TransactionType } from "../utils/constants";
+import { useNotificationManager } from "../hooks/useNotificationManager";
+import { Skeleton } from "@mui/material";
+import SolanaCalls from "../utils/callData/solanaCalls";
+import { utils } from "starknetid.js";
+import useHasClaimSolSubdomain from "../hooks/useHasClaimSolSubdomain";
+import DoneFilledIcon from "../components/UI/iconsComponents/icons/doneFilledIcon";
+import theme from "../styles/theme";
+import DomainActions from "../components/solana/domainActions";
+import DiscountEndScreen from "../components/discount/discountEndScreen";
+import identityChangeCalls from "../utils/callData/identityChangeCalls";
+
+const Solana: NextPage = () => {
+  const { address: starknetAddress } = useAccount();
+  const { publicKey: solPublicKey, signMessage } = useWallet();
+  const { addTransaction } = useNotificationManager();
+  const [walletModalOpen, setWalletModalOpen] = useState<boolean>(false);
+  const [currentStep, setCurrentStep] = useState<number>(0);
+  const [snsDomains, setSnsDomains] = useState<string[]>([]);
+  const [selectedDomain, setSelectedDomain] = useState<SnsDomainData>();
+  const [hasLoadedSolDomain, setHasLoadedSolDomain] = useState<boolean>(false);
+  const [isTxModalOpen, setIsTxModalOpen] = useState(false);
+  const claimedDomains = useHasClaimSolSubdomain(
+    snsDomains,
+    starknetAddress as string
+  );
+  const [callData, setCallData] = useState<Call[]>([]);
+  const { writeAsync: execute, data: registerData } = useContractWrite({
+    calls: callData,
+  });
+  const [open, setOpen] = useState(true);
+
+  useEffect(() => {
+    const currentDate = new Date();
+    const timestamp = currentDate.getTime();
+
+    if (timestamp >= 1706878847 * 1000) {
+      setOpen(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (!starknetAddress) setCurrentStep(0);
+    else if (starknetAddress && !solPublicKey) setCurrentStep(1);
+    else if (starknetAddress && solPublicKey) setCurrentStep(2);
+  }, [starknetAddress, solPublicKey]);
+
+  // fetch sol domains from SNS api
+  useEffect(() => {
+    if (!solPublicKey) return;
+    fetch(`https://sns-api.bonfida.com/owners/${solPublicKey}/domains`)
```

Comment:
should we implement a similar services implementation like we started in quests? wdyt?